core/src/storage.cpp

```diff
@@ -190,6 +190,10 @@ void SubTrajectory::fillMessage(moveit_task_constructor_msgs::msg::Solution& msg
 	moveit_task_constructor_msgs::msg::SubTrajectory& t = msg.sub_trajectory.back();
 	SolutionBase::fillInfo(t.info, introspection);
 
+	const auto trajectory_execution_info =
+	    creator()->properties().get<TrajectoryExecutionInfo>("trajectory_execution_info");
+	t.execution_info = trajectory_execution_info.toMsg();
```

Comment:
This way the execution info cannot be set *per solution*, but it is entirely valid that the same stage would generate solutions with, e.g., a left or a right arm. Accordingly, the controllers that you would need to execute the solution varies per solution.

Reply:
Is there a solution available to pick the trajectory info per solution/ any need for it at the moment? In order to unblock this PR I think this could be addressed in a follow-up issue when there is the need to specify trajectory info per solution.

Reply:
We would need to register custom user callbacks to fill those fields in a dynamic fashion.